pkg/stash/with_etcd.go

```diff
@@ -0,0 +1,95 @@
+package stash
+
+import (
+	"context"
+	"time"
+
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/observ"
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/clientv3/concurrency"
+	"golang.org/x/sync/errgroup"
+)
+
+// WithEtcd returns a distributed singleflight
+// using an etcd cluster. If it cannot connect,
+// then it will panic
+func WithEtcd(endpoints []string) (Wrapper, error) {
+	const op errors.Op = "stash.WithEtcd"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	c, err := clientv3.New(clientv3.Config{
+		Endpoints:   endpoints,
+		DialTimeout: time.Second * 5,
+	})
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	var eg errgroup.Group
+	for _, ep := range endpoints {
+		eg.Go(func() error {
+			_, err := c.Status(ctx, ep)
+			return err
+		})
+	}
+	err = eg.Wait()
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	return func(s Stasher) Stasher {
+		return &etcd{c, s}
+	}, nil
+}
+
+type etcd struct {
```

Comment:
Could we use more verbose names for struct variables? In this case something like 
`type etcd struct {	
        client *clientv3.Client
	stasher Stasher
}`

Reply:
done